### ISSUE TITLE
Compile time warning on GPUs

### ISSUE THREAD

**Issue by keitat** [kokkos/kokkos#1792]
I have been using CUDA 9.2.88 (and gcc7.2.0 on power9) to build a program with Kokkos.  During linking, I saw many warning message indicating "sm_35".  I do not intend to build a program for old NVIDIA devices (I set -DKOKKOS_ARCH=Volta20).  Do you have any thoughts on the warning message?

`nvlink warning : SM Arch ('sm_35') not found in 'CMakeFiles/Sparten_unit_test.dir/unit_test/AsciiIOUnitTest.cpp.o`

**Comment by crtrott** [comment]
I think the link line might not have been appended with the correct architecture flag. Is this simple Kokkos CMake or Trilinos?

**Comment by dholladay00** [comment]
I also see this when **not** compiling for compute capability 3.5. I use kokkos cmake.

**Comment by crtrott** [comment]
Yeah cc 3.5 is most likely the default of nvcc when not specifying the architecture. I assume this issue pops up when enabling relocatable device code?

**Comment by crtrott** [comment]
This is probably a bug in our CMake logic of handling linker arguments.

**Comment by ibaned** [comment]
@keitat can you send your full CMake configuration?

**Comment by keitat** [comment]
> cmake ../sparten -DKOKKOS_ENABLE_SERIAL:BOOL=OFF -DKOKKOS_ENABLE_OPENMP:BOOL=OFF -DKOKKOS_ENABLE_PTHREAD:BOOL=OFF  -DKOKKOS_ENABLE_LIBRT:BOOL=OFF -DKOKKOS_ARCH=Volta70,Power9  -DCMAKE_BUILD_TYPE=Release -DKOKKOS_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON  -DKOKKOS_ENABLE_LIBRT:BOOL=OFF -DKOKKOS_ENABLE_DEPRECATED_CODE:BOOL=OFF -DCMAKE_CXX_EXTENSIONS:BOOL=ON -DKOKKOS_ENABLE_PROFILING:BOOL=OFF -DKOKKOS_ENABLE_CUDA:BOOL=ON -### DKOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE:BOOL=ON -DKOKKOS_ENABLE_CUDA_UVM:BOOL=OFF -DKOKKOS_ENABLE_CUDA_LAMBDA:BOOL=ON -DKOKKOS_ENABLE_CUDA_LDG_INTRINSIC:BOOL=ON

**Comment by ibaned** [comment]
confirmed that `-arch=sm_70` is not on the link line.

**Comment by cwpearson** [comment]
This was most likely fixed in https://github.com/kokkos/kokkos/pull/2400
